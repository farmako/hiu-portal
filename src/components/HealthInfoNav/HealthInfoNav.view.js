import React from 'react';
import * as PropTypes from 'prop-types';
import { IconButton } from '@material-ui/core';
import { ChevronLeft, ChevronRight } from '@material-ui/icons';
import HealthInfoNavStyles from './HealthInfoNav.style';

const HealthInfoNav = ({ dates, selectedDate, setSelectedDate }) => {
  function selectNextDate() {
    const nextIndex = dates.indexOf(selectedDate) + 1;
    setSelectedDate(dates[nextIndex]);
  }

  function selectPreviousDate() {
    const previousIndex = dates.indexOf(selectedDate) - 1;
    setSelectedDate(dates[previousIndex]);
  }

  return (
    <HealthInfoNavStyles>
      <div className="health-info-nav">
        <span className="date-navigator-heading">View by visit date</span>
        <div className="date-navigator">
          <IconButton
            disabled={
              dates && dates !== [] ? dates.indexOf(selectedDate) === 0 : false
            }
            type="button"
            className="icon-button"
            aria-label="search"
            theme="primary"
            size="small"
            onClick={() => selectPreviousDate(dates, selectedDate)}
          >
            <ChevronLeft variant="outlined" theme="primary" />
          </IconButton>
          <div className="date-text">{selectedDate}</div>
          <IconButton
            disabled={
              dates && dates !== []
                ? dates.indexOf(selectedDate) === dates.length - 1
                : false
            }
            type="button"
            className="icon-button"
            aria-label="search"
            theme="primary"
            size="small"
            onClick={() => selectNextDate(dates, selectedDate)}
          >
            <ChevronRight variant="outlined" theme="primary" />
          </IconButton>
        </div>
      </div>
    </HealthInfoNavStyles>
  );
};

HealthInfoNav.propTypes = {
  dates: PropTypes.arrayOf(PropTypes.string),
  selectedDate: PropTypes.string,
  setSelectedDate: PropTypes.func.isRequired,
};

HealthInfoNav.defaultProps = {
  dates: [],
  selectedDate: '',
};

export default HealthInfoNav;
