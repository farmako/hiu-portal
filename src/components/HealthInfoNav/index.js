import HealthInfoNav from './HealthInfoNav.view';

export default HealthInfoNav;
