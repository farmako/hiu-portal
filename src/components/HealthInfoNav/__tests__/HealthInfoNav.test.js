import React from 'react';
import { shallow } from 'enzyme';
import IconButton from '@material-ui/core/IconButton';
import HealthInfoNav from '../HealthInfoNav.view';

describe('HealthInfo', () => {
  const mockDates = [
    '22/1/2020',
    '3/1/2020',
    '5/11/2019',
    '2/11/2019',
    '1/2/2013',
    '1/12/2012',
  ];
  const setSelectedState = jest.fn();
  const wrapper = shallow(
    <HealthInfoNav
      dates={mockDates}
      setSelectedDate={setSelectedState}
      selectedDate="22/1/2020"
    />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show navigation icons and on click call function setSelectedState', () => {
    wrapper.find(IconButton).at(0).simulate('click');
    expect(setSelectedState).toHaveBeenCalledTimes(1);
    wrapper.find(IconButton).at(1).simulate('click');
    expect(setSelectedState).toHaveBeenCalledTimes(2);
  });

  it('should enable/disable icon for left navigation', () => {
    expect(wrapper.find(IconButton).at(0).props().disabled).toEqual(true);
    wrapper.setProps({ selectedDate: '2/11/2019' });
    expect(wrapper.find(IconButton).at(0).props().disabled).toEqual(false);
  });

  it('should enable/disable icon for right navigation', () => {
    expect(wrapper.find(IconButton).at(1).props().disabled).toEqual(false);
    wrapper.setProps({ selectedDate: '1/12/2012' });
    expect(wrapper.find(IconButton).at(1).props().disabled).toEqual(true);
  });
});
