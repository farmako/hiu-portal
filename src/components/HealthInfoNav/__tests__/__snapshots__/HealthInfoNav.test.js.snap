// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthInfo should render properly 1`] = `
<styled.div>
  <div
    className="health-info-nav"
  >
    <span
      className="date-navigator-heading"
    >
      View by recorded date
    </span>
    <div
      className="date-navigator"
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="search"
        className="icon-button"
        disabled={true}
        onClick={[Function]}
        size="small"
        theme="primary"
        type="button"
      >
        <ChevronLeftIcon
          theme="primary"
          variant="outlined"
        />
      </WithStyles(ForwardRef(IconButton))>
      <div
        className="date-text"
      >
        22/1/2020
      </div>
      <WithStyles(ForwardRef(IconButton))
        aria-label="search"
        className="icon-button"
        disabled={false}
        onClick={[Function]}
        size="small"
        theme="primary"
        type="button"
      >
        <ChevronRightIcon
          theme="primary"
          variant="outlined"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </div>
</styled.div>
`;
